src/Password/Argon2i.php

```diff
@@ -0,0 +1,195 @@
+     * Constructor
+     *
+     * @param array|Traversable $options
+     *
+     * @throws Exception\InvalidArgumentException
+     */
+    public function __construct($options = [])
+    {
+        if (PHP_VERSION_ID < 70200) {
+            throw new Exception\InvalidArgumentException(
+                'The Argon2i password hash is only nativly available on PHP7.2+'
+            );
+        }
+
+        if (empty($options)) {
+            return;
+        }
+
+        if ($options instanceof Traversable) {
+            $options = ArrayUtils::iteratorToArray($options);
+        }
+
+        if (! is_array($options)) {
+            throw new Exception\InvalidArgumentException(
+                'The options parameter must be an array or a Traversable'
+            );
+        }
+
+        foreach ($options as $key => $value) {
+            switch (strtolower($key)) {
+                case 'memory_cost':
+                    $this->setMemoryCost($value);
+                    break;
+                case 'time_cost':
+                    $this->setTimeCost($value);
+                    break;
+                case 'threads':
+                    $this->setThreads($value);
+                    break;
+            }
+        }
+    }
+
+    /**
+     * Returns the maximum memory (in kibibytes) that may be used to compute the Argon2 hash
+     *
+     * @return int
+     */
+    public function getMemoryCost()
+    {
+        return $this->memoryCost;
+    }
+
+    /**
+     * Sets the maximum memory (in kibibytes) that may be used to compute the Argon2 hash
+     *
+     * @param int $memoryCost
+     *
+     * @return Argon2i
+     */
+    public function setMemoryCost($memoryCost)
+    {
+        $this->memoryCost = $memoryCost;
+        return $this;
+    }
+
+    /**
+     * Returns the maximum amount of time it may take to compute the Argon2 hash
+     *
+     * @return int
+     */
+    public function getTimeCost()
+    {
+        return $this->timeCost;
+    }
+
+    /**
+     * Sets the maximum amount of time it may take to compute the Argon2 hash
+     *
+     * @param int $timeCost
+     *
+     * @return Argon2i
+     */
+    public function setTimeCost($timeCost)
+    {
+        $this->timeCost = $timeCost;
+        return $this;
+    }
+
+    /**
+     * Returns the number of threads to use for computing the Argon2 hash
+     *
+     * @return int
+     */
+    public function getThreads()
+    {
+        return $this->threads;
+    }
+
+    /**
+     * Sets the number of threads to use for computing the Argon2 hash
+     *
+     * @param int $threads
+     *
+     * @return Argon2i
+     */
+    public function setThreads($threads)
+    {
+        $this->threads = $threads;
+        return $this;
+    }
+
+    /**
+     * @param  string $password
+     * @throws Exception\RuntimeException
+     * @return string
+     */
+    public function create($password)
+    {
+        $options = [];
+
+        if ($this->memoryCost !== null) {
```

Comment:
If we use the default PHP values, we can omit this check with null values and always pass the `$options` to `password_hash`.